plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSource.java

```diff
@@ -230,4 +233,11 @@ private static Block getRowIndexColumn(long baseIndex, int size)
         }
         return new LongArrayBlock(size, Optional.empty(), rowIndices);
     }
+
+    @Override
+    public Metrics getMetrics()
+    {
+        return new Metrics(parquetReader.getCodecsUsed().stream()
```

Comment:
If we collect the info about codec in parquetReader constructor, we can just save a reference to that in this class's constructor and avoid creating a new map each time.